### PR TITLE
Firebase bootcamp

### PR DESCRIPTION
**Pull request by MariemBaccari**
Completes the Firebase bootcamp with testing using the Firebase emulator.

### CODE REVIEW

File: app/src/main/java/com/github/sgueissa/bootcamp/FirebaseBootcamp.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                else future.complete(dataSnapshot.getValue().toString());
+            }
+        }).addOnFailureListener(
+                new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        future.completeExceptionally(e);
+                    }
+                }
+        );
```
Comment (simone-kalbermatter) [inline app/src/main/java/com/github/sgueissa/bootcamp/FirebaseBootcamp.java:45]: Maybe we could use lambda expressions here to make it more concise.

File: app/src/main/java/com/github/sgueissa/bootcamp/FirebaseBootcamp.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        future.thenAccept(
+                value -> email.setText(value)
+        );
+    }
+
+    public void set(View view){
+        String email = ((TextView) findViewById(R.id.email_text)).getText().toString();
+        String phone = ((TextView) findViewById(R.id.phone_text)).getText().toString();
```
Comment (simone-kalbermatter) [inline app/src/main/java/com/github/sgueissa/bootcamp/FirebaseBootcamp.java:53]: Is it possible to save ``(TextView) findViewById(R.id.email_text)`` in a constant or method somehow? For example in line 29/30 we are doing the same thing. For now it's not that important but for later there might be a cleaner way if elements get reused a lot.

File: app/src/main/res/layout/activity_firebase_bootcamp.xml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".FirebaseBootcamp">
+    <Button
+        android:id="@+id/get_button"
+        android:layout_width="79dp"
+        android:layout_height="37dp"
+        android:text="GET"
```
[the remark this answers is not in the document]
Comment (MariemBaccari) [reply app/src/main/res/layout/activity_firebase_bootcamp.xml:12]: Yes, we won't be using these fields in the project but we'll keep string resources in mind.

File: app/src/main/java/com/github/sgueissa/bootcamp/FirebaseBootcamp.java (hunk 1 above)
Comment (MariemBaccari) [reply app/src/main/java/com/github/sgueissa/bootcamp/FirebaseBootcamp.java]: Addressed in commit [43f7f86]

File: app/src/main/java/com/github/sgueissa/bootcamp/FirebaseBootcamp.java (hunk 2 above)
Comment (MariemBaccari) [reply app/src/main/java/com/github/sgueissa/bootcamp/FirebaseBootcamp.java]: I pushed the changes in commit [43f7f86]. I replaced findViewById instances by binding classes (which were not included in the bootcamp). They are not yet needed for the tests.

**Review by simone-kalbermatter — APPROVED** [review]

**Pull request closed without merging**